keystrokes/src/components/Character/CharacterList.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/character.css';
+import styleGuide from '../../stylesheets/style';
+import { ref } from '../../config/constants';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+
+function CharacterList(props) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState()
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		console.log(props)
+		setCharacters(props.userInfo.characters || {});
+		setCurrentCharacter(props.userInfo.currentCharacter);
+	}, [props]);
+
+	const handleOpen = () => {
+		setOpen(true);
+	}
+
+	async function handleSubmit(fields) {
+		const {name, characterClass, spec, race} = fields
+		let charactersSnapShot = await ref.child("/Characters");
+		let doesCharacterExist = await charactersSnapShot.child(name).once('value');
+		if (doesCharacterExist.exists()) {
```

Comment:
instead of character names as keys might be worth using a generated key, either the ones they have in firebase or something like https://www.npmjs.com/package/shortid